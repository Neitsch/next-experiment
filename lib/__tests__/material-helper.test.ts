describe("Page Context", () => {
  beforeEach(() => {
    jest.resetModules();
  });
  it("creates new server", () => {
    jest.doMock("jss", () => ({
      SheetsRegistry: jest.fn(),
    }));
    jest.doMock("@material-ui/core/styles", () => ({
      createGenerateClassName: jest.fn(),
      createMuiTheme: jest.fn(),
    }));
    const ctx = require("../material-helper").getPageContext();
    expect(ctx).toMatchSnapshot();
    expect(ctx).not.toBe(require("../material-helper").getPageContext());
  });
  it("creates new client", () => {
    process.browser = true;
    jest.doMock("jss", () => ({
      SheetsRegistry: jest.fn(),
    }));
    jest.doMock("@material-ui/core/styles", () => ({
      createGenerateClassName: jest.fn(),
      createMuiTheme: jest.fn(),
    }));
    const ctx = require("../material-helper").getPageContext();
    expect(ctx).toMatchSnapshot();
  });
  it("creates reload client", () => {
    process.browser = true;
    global.__INIT_MATERIAL_UI__ = jest.fn();
    jest.doMock("jss", () => ({
      SheetsRegistry: jest.fn(),
    }));
    jest.doMock("@material-ui/core/styles", () => ({
      createGenerateClassName: jest.fn(),
      createMuiTheme: jest.fn(),
    }));
    const ctx = require("../material-helper").getPageContext();
    expect(ctx).toBe(global.__INIT_MATERIAL_UI__);
  });
});
