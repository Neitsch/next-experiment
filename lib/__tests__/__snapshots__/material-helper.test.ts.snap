// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Context creates new client 1`] = `
Object {
  "generateClassName": undefined,
  "sheetsManager": Map {},
  "sheetsRegistry": mockConstructor {},
  "theme": undefined,
}
`;

exports[`Page Context creates new server 1`] = `
Object {
  "generateClassName": undefined,
  "sheetsManager": Map {},
  "sheetsRegistry": mockConstructor {},
  "theme": undefined,
}
`;
